sklearn/utils/_repr_html/params.py

```diff
@@ -50,12 +128,26 @@ def _params_html_repr(params):
             ></i></td>
             <td class="param">{param_name}&nbsp;</td>
             <td class="value">{param_value}</td>
+            <td class="doc_link"><a class="sk-estimator-doc-link wider-link"
+                                 {doc_link}
+                                </a>
+            </td>
         </tr>
     """
-
     rows = [
-        ROW_TEMPLATE.format(**_read_params(name, value, params.non_default))
-        for name, value in params.items()
+        ROW_TEMPLATE.format(
+            param_type=_read_params(row, params[row], params.non_default)["param_type"],
+            param_name=_read_params(row, params[row], params.non_default)["param_name"],
+            param_value=_read_params(row, params[row], params.non_default)[
+                "param_value"
+            ],
+            doc_link=_doc_row(
+                params.estimator_type,
+                row,
+                _read_params(row, params[row], params.non_default)["param_name"],
+            ),
+        )
+        for row in params
     ]
```

Comment:
nitpick for readability and avoid repetitions. We could do a for loop to reuse the result of `_read_params`

Reply:
something like 
```py
rows = []
for row in params:
    param = _read_params(row, ...)
    row.append(
        ROW_TEMPLATE.format(**param, doc_link=_doc_row(params.estimator_class, row, param["param_name"])
    )
```